src/com/lftechnology/java/training/niraj/stemleaf/StemLeaf.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * 
+ */
+package com.lftechnology.java.training.niraj.stemleaf;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * Helper class for data representation with stem and leaf plot
+ * 
+ * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+ *
+ */
+public class StemLeaf {
+	private static final int DIVIDER = 10;
+	private int[] dataPoints;
+	private Map<Integer, List<Integer>> stemLeafInfo;
+
+	public StemLeaf(int[] dataPoints) {
+		stemLeafInfo = new HashMap<Integer, List<Integer>>();
+		setDataPoints(dataPoints);
+	}
+
+	public int[] getDataPoints() {
+		return dataPoints;
+	}
+
+	public void setDataPoints(int[] dataPoints) {
+		this.dataPoints = dataPoints;
+		Arrays.sort(this.dataPoints);
+	}
+
+	/**
+	 * Plots stem and leaf for the given data points
+	 * 
+	 * @author Niraj Rajbhandari<nirajrajbhandari@lftechnology.com>
+	 */
+	public void plotStemLeaf() {
+		setStemLeafInfo(dataPoints);
+		displayStemLeafPlot();
+	}
+
+	/**
+	 * Set all the information related to stem and leaf plotting
+	 * 
+	 * @param dataPoints
+	 *            {@link Integer[]}
+	 * @author Niraj Rajbhandari <nirajrajbahndari@lftechnology.com>
+	 */
+	public void setStemLeafInfo(int[] dataPoints) {
+		int quotient;
+		int remainder;
+		float divisionResult;
+		for (int i = 0; i < dataPoints.length; i++) {
+			remainder = dataPoints[i] % DIVIDER;
+			divisionResult = dataPoints[i] / DIVIDER;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Cast one of the operands of this division operation to a "float". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2184)